dashboard/app/jobs/aichat_request_chat_completion_job.rb

```diff
@@ -87,8 +87,44 @@ def perform(request:, locale:)
     [SharedConstants::AI_REQUEST_EXECUTION_STATUS[:SUCCESS], response]
   end
 
+  # Checks for toxicity in the given text using various services, determined by DCDO settings.
+  # Returns {text: input (string), blocked_by: serviced that detected toxicity (string), details: filtering details (hash)}
+  private def find_toxicity(text, threshold, locale)
+    if blocklist_enabled?
+      text.split.each do |word|
+        return {text: text, blocked_by: 'blocklist', details: {blocked_word: word}} if profane_word_blocklist.include? word
+      end
+    end
+
+    if webpurify_enabled?
```

Comment:
Good question, right now it will always proceed in this order if all services are turned on. I figured we would proceed from most discrete/binary (i.e. it's in the list or it's not) to most open-ended (comprehend), but can confirm with Sam if we'd ever want to switch around the order.